File: Infrastructure/Mappers/ProductMapper.cs
using Domain.Models;
using Domain.ValueObjects.Product;
using Domain.ValueObjects.Shared;
using Infrastructure.DbEntities;

namespace Infrastructure.Mappers;

public static class ProductMapper
{
    public static ProductDbEntity FromDomainToDbEntity(Product domain)
    {
        return new ProductDbEntity(
            id: domain.Id.Value, 
            dateCreated: domain.DateCreated, 
            name: domain.Name,
            description: domain.Description,
            price: domain.Price.Value
        ) {
            Images = domain.Images.Select(ProductImageMapper.ToDbModel).ToList()
        };
    }

    public static Product FromDbEntityToDomain(ProductDbEntity dbEntity)
    {
        return new Product(
            id: ProductId.ExecuteCreate(dbEntity.Id), 
            name: dbEntity.Name,
            price: Money.ExecuteCreate(dbEntity.Price),
            description: dbEntity.Description,
            dateCreated: dbEntity.DateCreated,
            images: dbEntity.Images.Select(ProductImageMapper.ToDomain).ToList()
        );
    }
}

File: frontend/src/infrastructure/mappers/productMapper.ts
import IProduct from "../../domain/models/IProduct";
import IProductApiModel from "../apiModels/IProductApiModel";
import productImageMapper from "./productImageMapper";

const productMapper = {
    apiToDomain: (source: IProductApiModel): IProduct => {
        return {
            id: source.id,
            name: source.name,
            price: source.price,
            description: source.description,
            dateCreated: source.dateCreated,
            images: source.images.map(productImageMapper.apiToDomain),
        };
    },
};

export default productMapper;
